### PR TITLE
#161623847 Verify email via clicked link

### PR DESCRIPTION
**Pull request by KvNGCzA**
#### What does this PR do?

have `api/v1/users/verify` route working

#### Description of Task to be completed?

- add /api/v1/users/verify to users.js
- add tests for /api/v1/users/verify
- add verify JWT token function
- correct verify email message
- correct wrong userControllers test
- correct facebookLogin test

#### How should this be manually tested?

- `npm run start` to start server
- register for an account
- go to your email and click the link

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#161623847](https://www.pivotaltracker.com/story/show/161623847)

#### Screenshots (if appropriate)

Should successfully verify a registered user
screen shot 2018-11-06 at 8 26 35 pm

Should fail to verify an already verified user
screen shot 2018-11-06 at 8 26 44 pm

should fail to verify an unregistered user
screen shot 2018-11-06 at 8 27 06 pm

#### Questions:

N/A

### CODE REVIEW

File: server/controllers/UsersController.js [hunk 1]
```diff
@@ -92,12 +93,57 @@ class UsersController {
...
+    * @memberOf UserController
+    * @static
+    */
+  static verifyUser(req, res) {
+    const { token } = req.query;
+    const { id } = verifyToken(token);
+    User
+      .findByPk(id)
+      .then((user) => {
+        if (user.confirmEmail !== false) {
```
Comment (andela-moe) [inline server/controllers/UsersController.js]: I believe doing this 
```
if (confirmEmail) {
   console.log('this is true)
}
``` 

is same as what you have done unless you are trying to check for a specific value. Can you provide more clarity on your approach?

File: server/helpers/verifyEmailMessage.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const verifyEmailMessage = (token) => {
     `<div style="height: 20em, background-color: #E6FFED; 
       border: 1px solid black; padding: 0.5em;">
         <p>
-          <a href="http://localhost:3001/api/test/verify?token=${token}">
+          <a href="http://localhost:3001/api/v1/users/verify?token=${token}">
```
Comment (andela-moe) [inline server/helpers/verifyEmailMessage.js]: I believe that building out the URL this way would pose a problem later on. Please meet with @Darthrighteous He seems to have a similar acceptable solution for this.

Comment (KvNGCzA) [reply server/helpers/verifyEmailMessage.js]: @andela-moe thanks. I'll get on this.

File: server/helpers/tokenUtils.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export const createToken = (id, lifeSpan) => jwt
+  .sign({ id }, secret, { expiresIn: lifeSpan });
+
+/**
+ * Verifies the jwt and returns the payload
+ * @param {string} token - The jwt to be verified
+ * @returns {object} object containing user Id
+ */
+export const verifyToken = (token) => {
```
Comment (mbilesanmi) [inline server/helpers/tokenUtils.js]: This function seems to not be necessary. We can move the try-catch into the verifyJWT function in the middleware.

Comment (mbilesanmi) [reply server/helpers/tokenUtils.js]: Or better still combine this file and the verifyJWT middleware file into one.

Comment (Darthrighteous) [reply server/helpers/tokenUtils.js]: @mbilesanmi would that mean the `createToken` function too would be in the new file? I'm asking cause its a helper function, so we would have the `createToken` helper and the `verifyJWT` middleware in one file?

Comment (KvNGCzA) [reply server/helpers/tokenUtils.js]: @mbilesanmi can we discuss this during TTL interaction?

**Comment by KvNGCzA** [comment]
@mbilesanmi all changes made

**Comment by mrSerious** [comment]
LGTM

File: server/helpers/verifyEmailMessage.js [hunk 4]
```diff
@@ -16,7 +24,7 @@ const verifyEmailMessage = (token) => {
     `<div style="height: 20em, background-color: #E6FFED; 
       border: 1px solid black; padding: 0.5em;">
         <p>
-          <a href="http://localhost:3001/api/test/verify?token=${token}">
+          <a href="${protocol}://${address}/api/v1/users/verify?token=${token}">
```
Comment (mbilesanmi) [inline server/helpers/verifyEmailMessage.js]: You can change this to ""
```suggestion
          <a href="${process.env.API_BASE_URL}users/verify?token=${token}">
```
Then have the `http://localhost/api/v1/` in the env file.

Comment (mbilesanmi) [reply server/helpers/verifyEmailMessage.js]: That way we can avoid those plenty checks above.

File: package.json [hunk 5]
```diff
@@ -18,6 +18,7 @@
     "db-migrate": "sequelize db:migrate",
     "db-undo-migrate": "sequelize db:migrate:undo:all",
     "db-role-seed": "sequelize db:seed:all",
+    "db-init-test": "sequelize db:migrate:undo:all --env test && sequelize db:migrate --env test",
```
Comment (mbilesanmi) [inline package.json]: Are we still using this?

Comment (KvNGCzA) [reply package.json]: @mbilesanmi we don't. Removed

**Pull request merged**